src/plugins/template/tests/functional/op_reference/pad.cpp

```diff
@@ -1003,6 +1004,223 @@ std::vector<PadParams> generateParams() {
+            reference_tests::Tensor(ET_INT, {3}, std::vector<T_INT>{-1, 0, 0}),
+            reference_tests::Tensor(ET, {1, 2, 3}, std::vector<T>{
+                1, 2, 3,
+                4, 5, 6,
+            }),
+            op::PadMode::CONSTANT,
+            reference_tests::Tensor(ET, {}, std::vector<T>{0}),
+            "pad_neg_2x2x3_mode_const_remove_last"),
+        PadParams(
+            reference_tests::Tensor(ET, {2, 2, 3}, std::vector<T>{
+                1, 2, 3,
+                4, 5, 6,
+
+                7, 8, 9,
+                10, 11, 12
+            }),
+            reference_tests::Tensor(ET_INT, {3}, std::vector<T_INT>{-1, 0, 0}),
+            reference_tests::Tensor(ET_INT, {3}, std::vector<T_INT>{0, 0, 0}),
+            reference_tests::Tensor(ET, {1, 2, 3}, std::vector<T>{
+                7, 8, 9,
+                10, 11, 12
+            }),
+            op::PadMode::CONSTANT,
+            reference_tests::Tensor(ET, {}, std::vector<T>{0}),
+            "pad_neg_2x2x3_mode_const_remove_first"),
+        PadParams(
+            reference_tests::Tensor(ET, {3, 4}, std::vector<T>{
+                1, 2, 3, 4,
+                5, 6, 7, 8,
+                9, 10, 11, 12
+            }),
+            reference_tests::Tensor(ET_INT, {2}, std::vector<T_INT>{-1, 0}),
+            reference_tests::Tensor(ET_INT, {2}, std::vector<T_INT>{0, -2}),
+            reference_tests::Tensor(ET, {2, 2}, std::vector<T>{
+                5, 6,
+                9, 10
+            }),
+            op::PadMode::CONSTANT,
+            reference_tests::Tensor(ET, {}, std::vector<T>{0}),
+            "pad_neg_mix_cross_3x4_mode_const"),
+        PadParams(
+            reference_tests::Tensor(ET, {3, 4}, std::vector<T>{
+                1, 2, 3, 4,
+                5, 6, 7, 8,
+                9, 10, 11, 12
+            }),
+            reference_tests::Tensor(ET_INT, {2}, std::vector<T_INT>{-1, -2}),
+            reference_tests::Tensor(ET_INT, {2}, std::vector<T_INT>{2, 3}),
+            reference_tests::Tensor(ET, {4, 5}, std::vector<T>{
+                 7, 8, 0, 0, 0,
+                 11, 12, 0, 0, 0,
+                 0, 0, 0, 0, 0,
+                 0, 0, 0, 0, 0
+            }),
+            op::PadMode::CONSTANT,
+            reference_tests::Tensor(ET, {}, std::vector<T>{0}),
+            "pad_neg_begin_3x4_mode_const"),
+        PadParams(
+            reference_tests::Tensor(ET, {3, 4}, std::vector<T>{
+                1, 2, 3, 4,
+                5, 6, 7, 8,
+                9, 10, 11, 12
+            }),
+            reference_tests::Tensor(ET_INT, {2}, std::vector<T_INT>{2, 3}),
+            reference_tests::Tensor(ET_INT, {2}, std::vector<T_INT>{-1, -2}),
+            reference_tests::Tensor(ET, {4, 5}, std::vector<T>{
+                0, 0, 0, 0, 0,
+                0, 0, 0, 0, 0,
+                0, 0, 0, 1, 2,
+                0, 0, 0, 5, 6,
+                0, 0, 0, 9, 10,
```

Comment:
It 5x5 tensor, not 4x5. Remove `0,0,0,9,10` what is dropped. Did this test passed?